### DISCUSSION TITLE
Detect map changes

### DISCUSSION THREAD

**Discussion by ghost** [mrbungle64/ecovacs-deebot.js#284] [Q&A]
Is there a way to detect a map was changed?

**Comment by mrbungle64** [comment]
@cactusss 

You mean how to detect if the active map has changed?

**Comment by ghost** [comment]
No, I ment when a user makes changes to the map. I need to trigger a function in such a case, or else they will need to restart the app

Sent from Android device

Op 12 jan. 2023 08:55 schreef Sascha ***@***.***>:

@cactusss<https://github.com/cactusss>

You mean how to detect if the active map has changed?

**Reply by mrbungle64** [reply]
@cactusss 

Can you please give me an example in practice. In my experience, any change that is relevant triggers an event.

**Reply by mrbungle64** [reply]
@cactusss 

You can also run the `GetMaps` command if the bot starts charging, and you can also implement a function to sync the map data manually if you want to make sure.

**Reply by boriswerner** [reply]
@mrbungle64 I remember that we had one case where no event was being sent: on deletion of a virtual boundary. All other events were triggering an event. I just checked our chat in the iobroker forum and we discussed the topic on 7.12.2020. Perhaps this is still the case?

**Reply by mrbungle64** [reply]
@boriswerner 

Thanks for pointing this out 👍🏻 
I just checked it with the Deebot T8 AIVI, both creating and deleting a virtual boundary (via mobile app) did not trigger any event.

**Reply by ghost** [reply]
> @cactusss
> 
> Can you please give me an example in practice. In my experience, any change that is relevant triggers an event.

Sure. In my app I offer the users to clean multiple rooms. That can be saved in a flow that (or instance) gets triggered when the last persoon leaves home. Within that flow the roomnumbers are saved.
When a users changes the map (let's say divide or merge a room) those roomnumbers are not valid anymore.
If I can detect a map was changed I can run my procedure again to get the roomnumbers, and validate the flow.

Of course I can schedule to run a GetMaps periodically, but in theory that might fail when the map was changed and the schedule hasn't run yet.

Ans I can do it every minuten, but since I already get CPU warnings as it is I want to keep any unnecessary call to a minimum :)

**Reply by mrbungle64** [reply]
@cactusss 

Have you already pushed your latest changes to GitHub?
In order for me to give you proper tips it would be helpful if you would keep your repo up to date 😉 
I'm sure that we can find a solution for this challenge.

It is also always wise to take small steps rather than trying to address more complex challenges right at the beginning 😉

**Reply by ghost** [reply]
I'm new to github, other then browsing repositories from others I never used it. I just pushed a new version, well, I created a new repository.
